Sources/SwiftDocC/Model/Rendering/RenderNodeTranslator.swift

```diff
@@ -629,12 +629,16 @@ public struct RenderNodeTranslator: SemanticVisitor {
         collectedTopicReferences.append(contentsOf: hierarchyTranslator.collectedTopicReferences)
         node.hierarchyVariants = hierarchyVariants
         
-        // Emit variants only if we're not compiling an article-only catalog to prevent renderers from
-        // advertising the page as "Swift", which is the language DocC assigns to pages in article only catalogs.
-        // (github.com/swiftlang/swift-docc/issues/240).
-        if let topLevelModule = context.soleRootModuleReference,
-           try! context.entity(with: topLevelModule).kind.isSymbol
-        {
+        // If we're compiling an article-only catalog with a single root module
+        // that the articles are registered with, we do not want to create
+        // variants, to ensure that renderers don't advertise the page as
+        // "Swift", which is the language DocC assigns to pages in article only
+        // catalogs. (github.com/swiftlang/swift-docc/issues/240).
+        let skipVariants = context.soleRootModuleReference.flatMap {
+            // Ensure the root module's node is of a non-symbol type
+            try? !context.entity(with: $0).kind.isSymbol
+        } ?? false
+        if !skipVariants {
```

Comment:
I think this expression would be easier to follow without the double negation (`!` and "skip"). Renaming the variable to describe what it's checking could help make both the variable expression and the if-statement clearer.

We can also use `DocumentationContext.isSymbol(reference:)` to simplify the check and `context.rootModules.allSatisfy { ... }` if we don't need to verify that the catalog has a single root (which is the officially supported setup):

```suggestion
        let isArticleOnlyCatalog = context.rootModules.allSatisfy { context.isSymbol(reference: $0) == false }
        if !isArticleOnlyCatalog {
```